### ISSUE TITLE
[FEAT] Improvements to message commands

### ISSUE THREAD

**Issue by AstreaTSS** [NAFTeam/NAFF#392]
**Is your feature request related to a problem? Please describe.**
Message commands are limited, honestly. They have many problems, least of which is the fact that multiple arguments with differing types is a pain and subcommands are basically not happening with them.

**Describe the solution you'd like**
A system very much like `discord.py`'s message commands: a nice, mostly sane system to do everything a person would need.

```python
@message_command()
async def command(self, ctx, num: int, user: User):
    ...
```

**Describe alternatives you've considered**
Doing it yourself with the command arguments, which isn't pleasant, speaking from experience.
It's about about time we pointed out [`molter`](https://github.com/Discord-Snake-Pit/molter) exists, but `molter` will have to do more and more hacks to get things working as time goes on, and not every new user will know what `molter` is.

**Additional information**
See [`molter`'s #3](https://github.com/Discord-Snake-Pit/molter/issues/3) for what needs to be done to merge `molter` here, which would solve the issue.

*Under no circumstances should this issue hold back the beta release.* Polls and I have agreed it will not count towards it. It is under the milestone anyways to place a soft-soft deadline on myself.

**Comment by AstreaTSS** [comment]
Gonna add a note: this would be the *perfect* time to switch to the "prefixed commands" terminology that has been in use more and more frequently in the Discord bot development space. It helps separate them from context menu message commands while still being mostly clear.

**Comment by AstreaTSS** [comment]
Closed by #444.
